llms/mlx_lm/models/cache.py

```diff
@@ -126,6 +129,124 @@ def is_trimmable(self):
         return False
 
 
+def quantized_scaled_dot_product_attention(
```

Comment:
I would probably move this to `base.py` and replace it with a function called `scaled_dot_product_attention` that takes the same arguments as `mx.fast.scaled_dot_product_attention` plus the `cache`, then it routes to this or the `mx.fast` one.

This will also simplify the implementations a lot as the if goes in here and the models don't really care about the type of cache they are using.

Reply:
Great call! I've updated it everywhere in `mlx_lm`